sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -0,0 +1,486 @@
+
+    def resource(self):
+        """Return the built resource.
+        """
+        return self._parse_resource(self._pipeline_response)
+
+    @property
+    def _transport(self):
+        return self._client._pipeline._transport  # pylint: disable=protected-access
+
+    def initialize(self, client, initial_response, deserialization_callback):
+        """Set the initial status of this LRO.
+
+        :param initial_response: The initial response of the poller
+        :raises: HttpResponseError if initial status is incorrect LRO state
+        """
+        self._client = client
+        self._pipeline_response = self._initial_response = initial_response
+        self._deserialization_callback = deserialization_callback
+
+        for operation in self._lro_algorithms:
+            if operation.can_poll(initial_response):
+                self._operation = operation
+                break
+        else:
+            raise BadResponse("Unable to find status link for polling.")
+
+        try:
+            _raise_if_bad_http_status_and_method(self._initial_response.http_response)
+            self._status = self._operation.set_initial_status(initial_response)
+
+        except BadStatus as err:
+            self._status = 'Failed'
+            raise HttpResponseError(response=initial_response.http_response, error=err)
+        except BadResponse as err:
+            self._status = 'Failed'
+            raise HttpResponseError(response=initial_response.http_response, message=str(err), error=err)
+        except OperationFailed as err:
+            raise HttpResponseError(response=initial_response.http_response, error=err)
+
+    def run(self):
+        try:
+            self._poll()
+        except BadStatus as err:
+            self._status = 'Failed'
+            raise HttpResponseError(response=self._pipeline_response.http_response, error=err)
+
+        except BadResponse as err:
+            self._status = 'Failed'
+            raise HttpResponseError(response=self._pipeline_response.http_response, message=str(err), error=err)
+
+        except OperationFailed as err:
+            raise HttpResponseError(response=self._pipeline_response.http_response, error=err)
+
+    def _poll(self):
+        """Poll status of operation so long as operation is incomplete and
+        we have an endpoint to query.
+
+        :param callable update_cmd: The function to call to retrieve the
+         latest status of the long running operation.
+        :raises: OperationFailed if operation status 'Failed' or 'Canceled'.
+        :raises: BadStatus if response status invalid.
+        :raises: BadResponse if response invalid.
+        """
+
+        while not self.finished():
+            self._delay()
+            self.update_status()
+
+        if failed(self.status()):
+            raise OperationFailed("Operation failed or canceled")
+
+        if self._operation.should_do_final_get():
+            request = self._initial_response.http_response.request
```

Comment:
Are we not following the final `resourceLocation` attribute (in the response payload) if present?

Reply:
didn't know that was a thing....

Reply:
It's not in ARM:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#operation-resource-format

So didn't know the original spec has something like that, will update for azure-core

Reply:
fixed